pandas/core/groupby/indexing.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import annotations
+
+from typing import (
+    Iterable,
+    cast,
+)
+
+import numpy as np
+
+from pandas._typing import (
+    FrameOrSeries,
+    PositionalIndexer,
+)
+from pandas.util._decorators import (
+    cache_readonly,
+    doc,
+)
+
+from pandas.core.dtypes.common import (
+    is_integer,
+    is_list_like,
+)
+
+from pandas.core.groupby import groupby
+from pandas.core.indexes.api import CategoricalIndex
+
+
+class GroupByIndexingMixin:
+    """
+    Mixin for adding .rows to GroupBy.
+    """
+
+    @property
+    def _rows(self) -> _rowsGroupByIndexer:
+        return _rowsGroupByIndexer(cast(groupby.GroupBy, self))
+
+
+@doc(GroupByIndexingMixin._rows)
+class _rowsGroupByIndexer:
+    def __init__(self, grouped: groupby.GroupBy):
+        self.grouped = grouped
```

Comment:
can the name here be more obvious?  i had to double-check that this didnt correspond to `grouped.obj`

Reply:
I really struggled with that! I have renamed it groupByObject, which hopefully sums it up...

Reply:
Now it is renamed `groupby_object`.